File: samples/java-sample/src/main/java/com/lowbudgetlcs/riot4k/samples/java/AccountLookup.java
package com.lowbudgetlcs.riot4k.samples.java;

/** Application-level outcome of an account lookup. */
public sealed interface AccountLookup {
    record Found(String puuid, String riotId) implements AccountLookup {}

    record Missing() implements AccountLookup {}

    record Error(String message) implements AccountLookup {}
}

File: samples/jvm-sample/src/main/kotlin/com/lowbudgetlcs/riot4k/samples/jvm/AccountLookup.kt
package com.lowbudgetlcs.riot4k.samples.jvm

/** Application-level outcome of an account lookup. */
sealed interface AccountLookup {
    data class Found(val puuid: String, val riotId: String) : AccountLookup
    data object Missing : AccountLookup
    data class Error(val message: String) : AccountLookup
}
